libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1021,6 +1021,29 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			ArrayUtils(m_context).popStorageArrayElement(*arrayType);
 			break;
 		}
+		case FunctionType::Kind::BytesConcat:
+		{
+			_functionCall.expression().accept(*this);
+			vector<Type const*> argumentTypes;
+			for (auto const& argument: arguments)
+			{
+				argument->accept(*this);
+				solAssert(argument->annotation().type, "");
+				argumentTypes.emplace_back(argument->annotation().type);
+			}
+			utils().fetchFreeMemoryPointer();
+			// stack: <arg1> <arg2> ... <argn> <free mem>
+			m_context << u256(32) << Instruction::ADD;
+			utils().packedEncode(argumentTypes, vector<Type const*>{});
```

Comment:
Should we also do some sanity checks on the target types?

Reply:
Do you mean that I construct `targetTypes` here instead of passing empty vector?

Reply:
I think FixedBytes should be preferred if it works.

Reply:
I added same as I did in `IR` implementation.
If that is not good enough, I can add more checks